metricbeat/mb/testing/testdata.go

```diff
@@ -346,6 +361,12 @@ func documentedFieldCheck(foundKeys common.MapStr, knownKeys map[string]interfac
 					found = true
 					break
 				}
+				// should cover scenarios as azure.compute_vm_scaleset.*.*
+				key = strings.Join(splits[0:pos], ".") + ".*.*"
+				if _, ok := knownKeys[key]; ok {
+					found = true
+					break
+				}
```

Comment:
this should not be done in all values of `pos`, `.*.*` should only replace the last 2 splits. I think this code can go outside this loop